inc/macro_utils/macro_utils.h

```diff
@@ -97,82 +97,136 @@ MU_IF(X, "true", "false") => "true"
 /*MU_DEFINE_ENUM_WITHOUT_INVALID goes to header*/
 #define MU_DEFINE_ENUM_WITHOUT_INVALID(enumName, ...) typedef enum MU_C2(enumName, _TAG) { MU_FOR_EACH_1(MU_DEFINE_ENUMERATION_CONSTANT, __VA_ARGS__)} enumName; \
     typedef enum MU_C3(enum_value_metadata_, enumName, _TAG) { MU_FOR_EACH_1(MU_DEFINE_METADATA_ENUMERATION_VALUE, __VA_ARGS__) MU_C3(enum_value_metadata_, enumName, _VALUE_COUNT) = MU_COUNT_ARG(__VA_ARGS__)} MU_C2(enum_value_metadata_, enumName); \
-    extern const char* MU_C2(enumName,Strings)(enumName value); \
-    extern int MU_C2(enumName, _FromString)(const char* enumAsString, enumName* destination);
+    MU_DECLARE_ENUM_TO_STRING(enumName, __VA_ARGS__) \
 
 // this macro is a wrapper on top of MU_DEFINE_ENUM_WITHOUT_INVALID, adding an _INVALID value as the first enum value in the enum
 #define MU_DEFINE_ENUM(enumName, ...) \
     MU_DEFINE_ENUM_WITHOUT_INVALID(enumName, MU_C2(enumName, _INVALID), __VA_ARGS__)
 
+#define MU_INTERNAL_DEFINE_ENUM_VAR(enumName, enumValue) \
+    static enumName MU_C2(my_, enumValue);
+
+#define MU_INTERNAL_ASSIGN_ENUM_VALUE(argCount, enumValue) \
```

Comment:
>argCount [](http://example.com/codeflow?start=38&length=8)

is it me, or ArgCount is not used in the expansion of this macro... ?

Reply:
It is used now.

---
In reply to: [632096613](https://github.com/Azure/macro-utils-c/pull/30#discussion_r632096613) [](ancestors = 632096613)